src/Magnum/DartIntegration/World.h

```diff
@@ -0,0 +1,240 @@
+
+Common usage is to create a `dart::simulation::WorldPtr` and then instantiate
+a `DartIntegration::World` by passing a parent @ref SceneGraph::AbstractFeature and
+the `dart::simulation::WorldPtr` to its constructor:
+
+@code{.cpp}
+dart::simulation::WorldPtr dartWorld = createWorldInDart();
+addDartSkeletonsToDartWorld();
+SceneGraph::Scene<SceneGraph::MatrixTransformation3D> scene;
+SceneGraph::Object<SceneGraph::MatrixTransformation3D> object{&scene};
+// object should never be a @ref SceneGraph::Scene
+auto world = std::make_shared<DartIntegration::World>(object, dartWorld);
+@endcode
+
+Only the `dart::simulation::World` can affect the transformation of the Magnum
+@ref DartIntegration::World and not the other way around. Of course, one can
+change things directly in `dart::simulation::World` and observe the changes in
+@ref DartIntegration::World. To update the world and get the updated shapes you
+can do the following:
+
+@code{.cpp}
+std::shared_ptr<DartIntegration::World> world = createDartIntegrationWorld();
+// Simulate with time step of 0.001 seconds
+world->world()->setTimeStep(0.001);
+
+for(UnsignedInt i = 0; i < simulationSteps; i++) {
+    world->step();
+    // update graphics at ~60Hz
+    // 15*0.001 ~= 60Hz
+    if(i%15 == 0) {
+        world->refresh();
+
+        // get unused/deleted shapes
+        std::vector<std::shared_ptr<Object>> unusedObjects = world->unusedObjects();
+
+        deleteObjectsFromScene(unusedObjects);
+
+        // get updated shapes;
+        // ones that either the materials or the meshes have changed
+        std::vector<std::shared_ptr<Object>> updatedObjects = world->updatedShapeObjects();
+
+        updateMeshesAndMaterials(updatedObjects);
+
+        // clear list of updated objects
+        world->clearUpdatedShapeObjects();
+    }
+}
+@endcode
+
+@experimental
+*/
+class MAGNUM_DARTINTEGRATION_EXPORT World {
+    public:
+         /**
+         * @brief Constructor
+         * @param object    Parent object
+         * @param skeleton  DART World shared pointer to parse
+         */
+        template<class T> explicit World(T& object, std::shared_ptr<dart::simulation::World> world): _object(object), _manager{MAGNUM_PLUGINS_IMPORTER_DIR}, _dartWorld(world) {
+            objectCreator = [](SceneGraph::AbstractBasicObject3D<Float>& parent) -> SceneGraph::AbstractBasicObject3D<Float>* {
+                return new T{static_cast<T*>(&parent)};
+            };
+            dartObjectCreator = [](SceneGraph::AbstractBasicObject3D<Float>& parent, dart::dynamics::BodyNode* body) -> std::shared_ptr<Object> {
+                return std::make_shared<Object>(static_cast<T&>(parent), body);
+            };
+            dartShapeObjectCreator = [](SceneGraph::AbstractBasicObject3D<Float>& parent, dart::dynamics::ShapeNode* node) -> std::shared_ptr<Object> {
+                return std::make_shared<Object>(static_cast<T&>(parent), node);
+            };
+
+            refresh();
+        }
+
```

Comment:
The class has some heavy members, I would add an explicit destructor here and `= default` it in the `*.cpp` file to speed up the compilation.

Similarly for the constructor -- delegate most of the initialization to a private non-templated constructor that has a definition in the `*.cpp` file.